src/runtime/auto-prompt-manager.js

```diff
@@ -272,7 +272,12 @@ export class AutoPromptManager {
       (clientConfig?.autoPromptConfig?.clientDisplayTrigger
         ?.displayDelaySeconds || 0) * SECOND_IN_MILLIS;
 
-    if (shouldShowAutoPrompt) {
+    const interventionDisplayedIsAutoPromptType =
+      this.interventionDisplayed_ === null ||
+      // Only need to check contributions because Subsciptions interventions
+      // will not be saved.
+      this.isContribution_({autoPromptType: this.interventionDisplayed_.type});
+    if (shouldShowAutoPrompt && interventionDisplayedIsAutoPromptType) {
```

Comment:
Can you elaborate more on the edge case?
For example, is it possible `shouldShowAutoPrompt` is `true` and `interventionDisplayed_` is survey? if so, what's expected to be but what's broken?

Reply:
`interventionDisplayedIsAutoPromptType` can evaluates to true when there's no intervention to be displayed. When `shouldShowAutoPrompt` is `false`, I think it should evaluate to `false` as well.
I think we should move `shouldShowAutoPrompt ` up in the part of `interventionDisplayedIsAutoPromptType`?

Reply:
If no intervention is displayed at all, the control sequence exits before it reaches this line:
https://github.com/subscriptions-project/swg-js/pull/2934/files#diff-c74361e6f3e0a3fc5042519e0ddf35b67dc861e52b588913d6ebcf696ff17f12R248

Are you suggesting the following?
```
const interventionDisplayedIsAutoPromptType = shouldShowAutoPromptType && (
    this.interventionDisplayed_ === null ||this.isAutoPromptType_(this.interventionDisplayed_.type));
if (interventionDisplayedIsAutoPromptType) {
```

Reply:
Spoke offline. The decision from `getAudienceActionType_` and the prompt set in `this.interventionDisplayed_` both honor the decision to display autoprompt type.

Reply:
Q. Can we use `potentialAction` to determine triggering decision from 2 flows is an non-autoprompttype audience action?